src/sink_http.c

```diff
@@ -0,0 +1,200 @@
+    {
+        if (config->extended_counters) {
+            char suffixed[base_len + 8];
+            strcpy(suffixed, full_name);
+            SUFFIX_ADD(".count", json_integer(counter_count(value)));
+            SUFFIX_ADD(".mean", json_real(counter_mean(value)));
+            SUFFIX_ADD(".stdev", json_real(counter_stddev(value))); /* stdev matches other output */
+            SUFFIX_ADD(".sum", json_real(counter_sum(value)));
+            SUFFIX_ADD(".sum_sq", json_real(counter_squared_sum(value)));
+            SUFFIX_ADD(".lower", json_real(counter_min(value)));
+            SUFFIX_ADD(".upper", json_real(counter_max(value)));
+            SUFFIX_ADD(".rate", json_real(counter_sum(value) / config->flush_interval));
+        } else {
+            json_object_set_new(obj, full_name, json_real(counter_sum(value)));
+        }
+        break;
+    }
+    case SET:
+        json_object_set_new(obj, full_name, json_integer(set_size(value)));
+        break;
+    case TIMER:
+    {
+        timer_hist *t = (timer_hist*)value;
+        char suffixed[base_len + 20];
+        strcpy(suffixed, full_name);
+        SUFFIX_ADD(".sum", json_real(timer_sum(&t->tm)));
+        SUFFIX_ADD(".sum_sq", json_real(timer_squared_sum(&t->tm)));
+        SUFFIX_ADD(".mean", json_real(timer_mean(&t->tm)));
+        SUFFIX_ADD(".lower", json_real(timer_min(&t->tm)));
+        SUFFIX_ADD(".upper", json_real(timer_max(&t->tm)));
+        SUFFIX_ADD(".count", json_integer(timer_count(&t->tm)));
+        SUFFIX_ADD(".stdev", json_real(timer_stddev(&t->tm))); /* stdev matches other output */
+        for (int i = 0; i < config->num_quantiles; i++) {
+            char ptile[20];
+            sprintf(ptile, ".p%0.0f", config->quantiles[i] * 100);
+            SUFFIX_ADD(ptile, json_real(timer_query(&t->tm, config->quantiles[i])));
+        }
+        SUFFIX_ADD(".rate", json_real(timer_sum(&t->tm) / config->flush_interval));
+        SUFFIX_ADD(".sample_rate", json_real((double)timer_count(&t->tm) / config->flush_interval));
+
+        /* Manual histogram bins */
+        if (t->conf) {
+            char ptile[20];
+            sprintf(ptile, ".bin_<%0.2f", t->conf->min_val);
+            SUFFIX_ADD(ptile, json_integer(t->counts[0]));
+            for (int i = 0; i < t->conf->num_bins - 2; i++) {
+                sprintf(ptile, ".bin_%0.2f", t->conf->min_val+(t->conf->bin_width*i));
+                SUFFIX_ADD(ptile, json_integer(t->counts[i+1]));
+            }
+            sprintf(ptile, ".bin_>%0.2f", t->conf->max_val);
+            SUFFIX_ADD(ptile, json_integer(t->counts[t->conf->num_bins - 1]));
+        }
+        break;
+    }
+    default:
+        syslog(LOG_ERR, "Unknown metric type: %d", type);
+        break;
+    }
+
+    return 0;
+}
+
+static int json_cb(const char* buf, size_t size, void* d) {
+    strbuf* sbuf = (strbuf*)d;
+    strbuf_cat(sbuf, buf, size);
+    return 0;
+}
+
+static int serialize_metrics(http_sink* sink, metrics* m, void* data) {
```

Comment:
the whole serialize workflow? Yeah...

A bit awkward with how the test suite is run right now, where testing is only possible on public interfaces. 

I'm going to add a second constructor interface so I don't leak all of the internal state to a test and check serialization output.